app/jobs/get_next_ai_message_job.rb

```diff
@@ -0,0 +1,23 @@
+class GetNextAiMessageJob < ApplicationJob
+  def perform(conversation_id)
+    puts "GetNextAiMessageJob.perform(#{conversation_id})"
+
+    @conversation = Conversation.find conversation_id
+    messages = @conversation.messages.sorted.map(&:for_openai)
+    @new_message = @conversation.messages.create! role: :assistant, content_text: "", assistant: @conversation.assistant
+    @new_message.broadcast_append_to @conversation
+
+    response = AiBackends::OpenAi.new(@conversation.user.openai_key)
+      .get_next_chat_message(messages) do |content_chunk|
+        @new_message.content_text += content_chunk
+        @new_message.broadcast_replace_to @new_message.conversation, locals: { scroll_into_view: true }
+      end
```

Comment:
@krschacht I'm actually pretty happy with this cleaner API. It's nice to be able to hide the mess of building the proc behind a veil.